### PR TITLE
Framework Escaper - remove direct Object Manager usage

### PR DESCRIPTION
**Pull request by maciejslawik**
### Description (*)

This PR removes direct ObjectManager calls from ```\Magento\Framework\Escaper```. Additionally, it allows for injecting different implementations of ```\Psr\Log\LoggerInterface``` via the constructor.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=19310) <br/>All committers have signed the CLA.

**Comment by magento-engcom-team** [comment]
Hi @maciejslawik. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by sivaschenko** [comment]
Thanks for your contribution @maciejslawik , however adding a constructor is backward incompatible change - it will result in issue on instances where this class is instantiated or extended. Please see https://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/

**Comment by maciejslawik** [comment]
@sivaschenko Alright, I understand. 
How about adding this constructor and setting default values for those parameters to null and keeping those 2 getters? This should provide backward compatibility and still give a chance to inject custom implementations.

File: lib/internal/Magento/Framework/Escaper.php [hunk 1]
```diff
@@ -37,6 +37,19 @@ class Escaper
...
+     * Escaper constructor.
+     * @param \Magento\Framework\ZendEscaper $escaper
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(
+        \Magento\Framework\ZendEscaper $escaper,
+        \Psr\Log\LoggerInterface $logger
+    ) {
+        $this->escaper = $escaper;
+        $this->logger = $logger;
```
Comment (orlangur) [inline lib/internal/Magento/Framework/Escaper.php]: Such change is backward-incompatible. 

Please inject new dependencies in backward-compatible manner: https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/#adding-a-constructor-parameter

**Review by orlangur — CHANGES_REQUESTED** [review]

File: lib/internal/Magento/Framework/Escaper.php (hunk 1 above)
Comment (maciejslawik) [reply lib/internal/Magento/Framework/Escaper.php]: I added null as default to keep all instances that are instantiated directly unbroken.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3518](https://jira.corp.magento.com/browse/ENGCOM-3518) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@maciejslawik thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: lib/internal/Magento/Framework/Escaper.php (hunk 1 above)
Comment (orlangur) [reply lib/internal/Magento/Framework/Escaper.php]: @maciejslawik please remove private deprecated methods as you did before and add property initialization into constructor (as in devdoc I mentioned).

**Review by orlangur — CHANGES_REQUESTED** [review]

**Comment by maciejslawik** [comment]
@orlangur I pushed changes that should reflect your request.

**Comment by orlangur** [comment]
@maciejslawik great! Changes look good now, please rebase against latest `2.3-develop` as there is a conflict.

**Comment by maciejslawik** [comment]
@orlangur Rebase has been done.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-3518](https://jira.corp.magento.com/browse/ENGCOM-3518) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
Hi @maciejslawik. Thank you for your contribution. 
We will aim to release these changes as part of 2.3.1. 
Please check the release notes for final confirmation.

**Comment by sivaschenko** [comment]
Hi @maciejslawik unfortunately we had to revert this PR changes because of backward compatibility concerns:
If the Escaper was extended and parent constructor was not called - that will result in `escaper` and `logger` properties not been inititalized

**Comment by orlangur** [comment]
@sivaschenko mentioned use case does not seem valid to me. It means any other similar addition of dependency is backward incompatible.

First of all, we do not recommend inheritance. In the second place, it is not correct to not call parent constructor.

**Comment by sivaschenko** [comment]
@orlangur in this specific case there is a high risk that if the class was extended the parent constructor was not called as there was no constructor implementation in place.

**Comment by orlangur** [comment]
@slitvyachenko did somebody really extend this class having some troubles? I believe we should follow https://devdocs.magento.com/guides/v2.3/contributor-guide/backward-compatible-development/ and nothing more than that. If doc is not sufficient, it should be extended with your note - "dependencies MUST NOT be added in case there were none before" or something.

**Comment by sivaschenko** [comment]
@orlangur I believe that change is falling into the category of "adding public method" rather than adding more dependencies (each class does have a constructor, but in this case, as the constructor was not explicitly defined, it would have been treated as it does not exist).

**Pull request merged**